standard/classes.md

```diff
@@ -2130,7 +2130,7 @@ default_argument
 
 parameter_modifier
     : parameter_mode_modifier
-    | 'this'
+    | 'this' parameter_mode_modifier?
```

Comment:
The ordering is flexible:

```suggestion
    | 'this' parameter_mode_modifier?
    | parameter_mode_modifier? 'this'
```

Reply:
My test shows that the order is indeed flexible. Given Nigel's proposal, `this` on its own would be ambiguous, but I think ANTLR chooses the lexical-first rule that matches, right, Nigel?

Reply:
@RexJaeschke – Yes, ANTLR resolves such ambiguity by picking the first alternative. We have noted elsewhere where the grammar relies on this, however in this case as the two alternatives represent exactly the same thing I don’t think a note is necessary.